packages/atomic/src/components/insight/atomic-insight-result-list/atomic-insight-result-list.tsx

```diff
@@ -0,0 +1,222 @@
+
+  /**
+   * The spacing of various elements in the result list, including the gap between results, the gap between parts of a result, and the font sizes of different parts in a result.
+   */
+  @Prop({reflect: true}) density: ResultDisplayDensity = 'normal';
+  /**
+   * The expected size of the image displayed in the results.
+   */
+  @Prop({reflect: true}) imageSize: ResultDisplayImageSize = 'icon';
+
+  // TODO:
+  // /**
+  //  * @internal
+  //  */
+  // @Method() public async focusOnNextNewResult() {
+  //   this.resultListCommon.focusOnNextNewResult(this.resultListState);
+  // }
+
+  public async initialize() {
+    this.resultList = buildInsightResultList(this.bindings.engine, {
+      options: {
+        fieldsToInclude: this.bindings.store.state.fieldsToInclude || undefined,
+      },
+    });
+    this.registerResultTemplates();
+    // TODO:
+    // this.resultsPerPage = buildResultsPerPage(this.bindings.engine);
+    // this.bindings.store.registerResultList(this);
+  }
+
+  public getTemplate(result: InsightResult) {
+    return this.resultTemplatesManager.selectTemplate(result);
+  }
+
+  private async registerResultTemplates() {
+    this.resultTemplatesManager = insightBuildResultTemplatesManager(
+      this.bindings.engine
+    );
+    const elements = this.host.querySelectorAll(
+      'atomic-insight-result-template'
+    ) as NodeListOf<TemplateElement>;
+    const customTemplates = await Promise.all(
+      Array.from(elements).map(async (resultTemplateElement) => {
+        const template = await resultTemplateElement.getTemplate();
+        if (!template) {
+          this.templateHasError = true;
+        }
+        return template;
+      })
+    );
+
+    const templates = [this.makeDefaultTemplate()].concat(
+      customTemplates.filter(
+        (template) => template
+      ) as InsightResultTemplate<DocumentFragment>[]
+    );
+
+    this.resultTemplatesManager.registerTemplates(...templates);
+    this.ready = true;
+  }
+
+  private makeDefaultTemplate(): InsightResultTemplate<DocumentFragment> {
+    const content = document.createDocumentFragment();
+    const linkEl = document.createElement('atomic-result-link');
+    content.appendChild(linkEl);
+    return {
+      content,
+      conditions: [],
+    };
+  }
+
+  private getContentOfResultTemplate(
+    result: InsightResult
+  ): HTMLElement | DocumentFragment {
+    return this.getTemplate(result)!;
+  }
+  private getClasses(): string {
+    const classes = getResultDisplayClasses(
+      'list',
+      this.density,
+      this.imageSize
+    );
+    if (
+      this.resultListState.firstSearchExecuted &&
+      this.resultListState.isLoading
+    ) {
+      classes.push('loading');
+    }
+    if (!this.bindings.store.isAppLoaded()) {
+      classes.push('placeholder');
+    }
+    return classes.join(' ');
+  }
+
+  render() {
+    if (!this.ready) {
+      return null;
+    }
+    return (
```

Comment:
Once those TODOs are completed, perhaps some of this could also be moved to the common result list somehow.